sdk/containerregistry/container-registry/src/blob/containerRegistryBlobClient.ts

```diff
@@ -0,0 +1,433 @@
+    const serviceVersion = options.serviceVersion ?? LATEST_API_VERSION;
+    const authClient = new GeneratedClient(endpoint, serviceVersion, internalPipelineOptions);
+    this.client = new GeneratedClient(endpoint, serviceVersion, internalPipelineOptions);
+    this.client.pipeline.addPolicy(
+      bearerTokenAuthenticationPolicy({
+        credential,
+        scopes: [defaultScope],
+        challengeCallbacks: new ChallengeHandler(
+          new ContainerRegistryRefreshTokenCredential(authClient, defaultScope, credential)
+        ),
+      })
+    );
+  }
+
+  /**
+   * Delete a blob.
+   * @param digest - the digest of the blob to delete
+   * @param options - optional configuration used to send requests to the service
+   */
+  public async deleteBlob(digest: string, options?: DeleteBlobOptions): Promise<void> {
+    const { span, updatedOptions } = createSpan("ContainerRegistryBlobClient-deleteBlob", options);
+
+    try {
+      await this.client.containerRegistryBlob.deleteBlob(
+        this.repositoryName,
+        digest,
+        updatedOptions
+      );
+    } catch (e) {
+      span.setStatus({ code: SpanStatusCode.ERROR, message: e.message });
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Upload a manifest for an OCI artifact.
+   *
+   * @param manifest - the manifest to upload
+   * @param options - optional configuration used to send requests to the service
+   */
+  public async uploadManifest(
+    manifest: OciManifest,
+    options?: UploadManifestOptions
+  ): Promise<UploadManifestResult>;
+
+  /**
+   * Upload a manifest for an OCI artifact.
+   *
+   * @param manifestStream - the raw manifest as a stream
+   * @param options - optional configuration used to send requests to the service
+   */
+  public async uploadManifest(
+    manifestStream: NodeJS.ReadableStream,
+    options?: UploadManifestOptions
+  ): Promise<UploadManifestResult>;
+
+  /**
+   * Upload a manifest for an OCI artifact.
+   *
+   * @param manifestStreamFactory - a function which returns a stream of the raw manifest. This function may be called multiple times; each time the function is called a fresh stream should be returned.
+   * @param options - optional configuration used to send requests to the service
+   */
+  public async uploadManifest(
+    manifestStreamFactory: () => NodeJS.ReadableStream,
+    options?: UploadManifestOptions
+  ): Promise<UploadManifestResult>;
+
+  /**
+   * Upload a manifest for an OCI artifact.
+   *
+   * @param manifest - the manifest to upload. If a resettable stream (a factory function that returns a stream) is provided, it may be called multiple times. Each time the function is called, a fresh stream should be returned.
+   */
+  public async uploadManifest(
+    manifest: (() => NodeJS.ReadableStream) | NodeJS.ReadableStream | OciManifest,
+    options?: UploadManifestOptions
+  ): Promise<UploadManifestResult> {
+    const { span, updatedOptions } = createSpan(
```

Comment:
sorry I didn't realize that we are still use old way of creating spans. Not urgent, but we want to use the new style

https://github.com/Azure/azure-sdk-for-js/blob/50659ce050eaae2d39d8e22d7cb1cbc57dd961f9/sdk/tables/data-tables/src/TableClient.ts#L578

If time is tight, we can do this in the next release.

Reply:
Will leave for next release. I'll log an issue and can update the rest of ACR at the same time.